workshop/6-CustomerDataPlatforms/6.1-Segment.ipynb

```diff
@@ -0,0 +1,303 @@
+    "\n",
+    "Next, we will use the aforementioned service to pull our demo user recommendations and push them into Segment. We’re going to install Segment’s Python library, then create demo user profiles within Segment and pull in recommendations for each user from Amazon Personalize:\n"
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import sys\n",
+    "!{sys.executable} -m pip install analytics-python\n",
+    "import analytics\n",
+    "\n",
+    "analytics.write_key = segment_write_key\n",
+    "\n",
+    "for id in range(10, 20):\n",
+    "    analytics.identify('personalize_user_'+str(id), {\n",
+    "        'email': 'personalize_user_'+str(id)+'@example.com',\n",
+    "        'name': 'Example User'+str(id)\n",
+    "    })\n",
+    "    response = requests.get('http://{}/recommendations?userID={}&fullyQualifyImageUrls=1&numResults=10'.format(recommendations_elb_domain_name, id))\n",
+    "    for product_recomendation in response.json():\n",
+    "        analytics.track('personalize_user_'+str(id), 'Product Recommended', {\n",
+    "            'product_id': product_recomendation['product']['id'],\n",
+    "            'product_url': product_recomendation['product']['url'],\n",
+    "            'product_name': product_recomendation['product']['name'],\n",
+    "            'product_category': product_recomendation['product']['category'],\n",
+    "            'product_style': product_recomendation['product']['style'],\n",
+    "            'product_description': product_recomendation['product']['description'],\n",
+    "            'product_price': product_recomendation['product']['price'],\n",
+    "            'product_image': product_recomendation['product']['image']\n",
+    "        })"
+   ]
+  },
+  {
+   "source": [
+    "## Create a user trait for each user in Segment to calculate the most recommended product category in the last 30 days\n",
+    "\n",
+    "Now that our users and their associated Amazon Personalize recommendations are in Segment, let’s inspect a user profile to understand the data structure. From the Segment Personas Explorer (/personas/explorer) click into one of the user profiles we’ve created and inspect the user traits and associated events we just created:\n",
+    "\n",
+    "![Segment Personas](images/segment/personas.png)\n",
+    "\n",
+    "In order to leverage these recommendations for omnichannel personalization, let’s add a trait to each user profile that calculates the most recommended category for each user in the last 30 days. Segment calls these Computed Traits. Computed Traits allow you to quickly create user or account-level calculations that Segment keeps up-to-date over time - learn more about Computed Traits here.\n",
```

Comment:
Would be a nice touch to hyperlink Segment concepts/features like "Computed Traits" to pages on the Segment website where attendees can learn more.